src/Components/TaskList.js

```diff
@@ -2,39 +2,51 @@ import React, { Component } from 'react';
 import TaskItem from './TaskItem';
 import SelectInput from './SelectInput';
 import { connect } from "react-redux";
+import * as actions from "../actions/index"
 
 class TaskList extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      filterName: ' ',
+      filterName: null,
       filterStatus: 0,
     }
   }
 
   changeValueFilterByValue = (e) => {
     const { filterName, filterStatus } = this.state;
     const { value, name } = e.target;
-
-    this.props.getValueFilter(
-      name === 'filterName' ? value : filterName,
-      name === 'filterStatus' ? value : filterStatus,
-    )
+    const filter = {
+      name: name === 'filterName' ? value : filterName,
+      status: name === 'filterStatus' ? Number(value) : filterStatus,
+    }
+    this.props.onFilter(filter)
     this.setState({
       [name]: value,
     })
   }
 
   render() {
-    const { data, editProduct } = this.props;
+    let { data, isFilterTable } = this.props;
+    if (isFilterTable) {
+      const { name, status } = isFilterTable;
+      if (name) {
+        data = data.filter(({ name : nameItem }) => nameItem.toLowerCase().indexOf(name) !== -1)
+      }
+
+      if (status) {
+        data = data.filter(({ status : nameStatus }) => {
+          if (status === 0) {
+            return data;
+          } else {
+            return nameStatus === (status === 1 ? false : true)
```

Comment:
```
return nameStatus === (!(status === 1 ))
```
nhưng đang thắc mắc sao status == 1 mà nó ko phải là kiểu boolean

Reply:
Cả cái cục từ dòng 32-->46 : Move sang 1 function khác 
Ex:
data = this.applyFilterData(data)